File: stdin.java
import java.util.ArrayList;
import java.util.Scanner;

class StdIn {
  public static void main(String args[]) {
    Scanner scanner = new Scanner(System.in);
    String input_text;
    ArrayList<String> lines = new ArrayList<String>();
    while (scanner.hasNextLine()) {
      input_text = scanner.nextLine();
      lines.add(input_text);
    }
    System.out.println(lines);
    scanner.close();
  }
}

File: stdin.scala
import java.util.Scanner
import scala.collection.mutable.ListBuffer
object Main extends App {
  val scanner = new Scanner(System.in)
  val lines: ListBuffer[String] = ListBuffer()
  while (scanner.hasNext) {
    lines += scanner.nextLine()
  }
  println(lines)
  scanner.close
}
